worker/mozillaEvaluationWorker/mozillaEvaluationWorker.go

```diff
@@ -0,0 +1,550 @@
+		status = false
+	}
+
+	if !hasOCSP {
+		failures = append(failures, "consider enabling OCSP stapling")
+	}
+
+	if !hasSHA1 {
+		failures = append(failures, "it is recommended to use sha1")
+		status = false
+	}
+
+	if !has3DES {
+		failures = append(failures, "add cipher DES-CBC3-SHA")
+		status = false
+	}
+
+	if !hasPFS {
+		failures = append(failures, "use DHE of at least 2048bits and ECC of at least 256bits")
+		status = false
+	}
+
+	return status, failures
+}
+
+func isIntermediate(c connection.Stored) (bool, []string) {
+
+	status := true
+	var allProtos []string
+	hasTLSv1 := false
+	hasAES := false
+	hasSHA256 := true
+	hasOCSP := true
+	hasPFS := true
+	var failures []string
+
+	for _, cs := range c.CipherSuite {
+
+		if !contains(intermediate.Ciphers, cs.Cipher) {
+			failures = append(failures, fmt.Sprintf("remove %s cipher", cs.Cipher))
+			status = false
+		}
+
+		for _, proto := range cs.Protocols {
+			if !contains(allProtos, proto) {
+				allProtos = append(allProtos, proto)
+			}
+		}
+
+		if contains(cs.Protocols, "TLSv1") {
+			hasTLSv1 = true
+		}
+
+		if cs.Cipher == "AES128-SHA" {
+			hasAES = true
+		}
+
+		if cs.PFS != "None" {
+			if !hasGoodPFS(cs.PFS, 2048, 256, false) {
+				hasPFS = false
+			}
+		}
+
+		if cs.SigAlg != intermediate.CertificateSignature {
+
+			fail := fmt.Sprintf("%s is not an intermediate signature", cs.SigAlg)
+			if !contains(failures, fail) {
+				failures = append(failures, fail)
+			}
+			hasSHA256 = false
+		}
+
+		if !cs.OCSPStapling {
+			hasOCSP = false
+		}
+	}
+
+	extraProto := extra(intermediate.TLSVersions, allProtos)
+	for _, p := range extraProto {
+		failures = append(failures, fmt.Sprintf("disable %s protocol", p))
+		status = false
+	}
+
+	if !hasAES {
+		failures = append(failures, "add cipher AES128-SHA")
+		status = false
+	}
+
+	if !hasTLSv1 {
+		failures = append(failures, "consider adding TLSv1")
+		status = false
+	}
+
+	if !c.ServerSide {
+		failures = append(failures, "enforce ServerSide ordering")
+	}
+
+	if !hasOCSP {
+		failures = append(failures, "consider enabling OCSP stapling")
+	}
+
+	if !hasSHA256 {
+		failures = append(failures, "it is recommended to use sha256")
+	}
+
+	if !hasPFS {
+		failures = append(failures, "use DHE of at least 2048bits and ECC of at least 256bits")
+	}
+
+	return status, failures
+}
+
+func isModern(c connection.Stored) (bool, []string) {
+
+	status := true
+	var allProtos []string
+	hasSHA256 := true
+	hasOCSP := true
+	hasPFS := true
+	var failures []string
+
+	for _, cs := range c.CipherSuite {
+
+		if !contains(modern.Ciphers, cs.Cipher) {
+			failures = append(failures, fmt.Sprintf("remove %s cipher", cs.Cipher))
+			status = false
+		}
+
+		for _, proto := range cs.Protocols {
+			if !contains(allProtos, proto) {
+				allProtos = append(allProtos, proto)
+			}
+		}
+
+		if cs.PFS != "None" {
+			if !hasGoodPFS(cs.PFS, 2048, 256, false) {
+				hasPFS = false
+			}
+		}
+
+		if cs.SigAlg != modern.CertificateSignature {
```

Comment:
Hadn't realised that.
Going to change it right away and hopefully we will finish with the worker.